src/main/java/com/alexlasota/medicalclinic/service/PatientService.java

```diff
@@ -0,0 +1,65 @@
+package com.alexlasota.medicalclinic.service;
+
+import com.alexlasota.medicalclinic.model.Password;
+import com.alexlasota.medicalclinic.model.Patient;
+import com.alexlasota.medicalclinic.repository.PatientRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class PatientService {
+
+    private final PatientRepository patientRepository;
+
+    public List<Patient> getPatients() {
+        return patientRepository.getPatients();
+    }
+
+    public Patient getPatientByEmail(String email) {
+        return patientRepository.getPatientByEmail(email)
+                .orElseThrow(() -> new IllegalArgumentException("Patient with given email doesnt exist"));
+    }
+
+    public void addPatient(Patient patient) {
+        patientRepository.addPatient(patient);
+    }
+
+    public void removePatientByEmail(String email) {
+        patientRepository.removePatientByEmail(email);
+    }
+
+    public Patient editPatient(String email, Patient newPatientData) {
+        Patient toEditPatient = patientRepository.getPatientByEmail(email)
+                .orElseThrow(() -> new IllegalArgumentException("Patient with given email doesnt exist"));
+
+        patientRepository.editPatient(toEditPatient, newPatientData);
+        return toEditPatient;
+    }
+    public Optional<Patient> updatePassword(String email, Patient newPassword) {
+        Optional<Patient> optionalPatient = Optional.ofNullable(getPatientByEmail(email));
```

Comment:
dlaczego uzywasz metody bezposrednio z serwisu? -> uzyj repozytorium aby pobrać pacjenta. Repozytorium powinno dostarczać Ci dane. 

Wtedy nie bedziesz musial opakowywac kolejny raz w optionala.

Nastepnie w sytuacji kiedy pacjenta takiego nie znaleziono to powinnismy rzucic wyjatek po co w ogole cokolwiek robic dalej jesli pacjent, ktoremu zostala zlecona zmiana hasla nie istnieje. 
Wtedy dalej nie bedziemy mieli zadnego sprawdzenia if isPresent()

Wystarczy, że sobie zmienimy haslo pacjentowi.